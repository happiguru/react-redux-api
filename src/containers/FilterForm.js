import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import axios from 'axios';
import setTitleFilter from '../actions/Filter';
import { addFoods } from '../actions/Foods';
import '../styles/filterform.css';

const FilterForm = ({ addFoods }) => {
  const [input, setInput] = useState('');
  const fetchData = async () => {
    try {
      const { data } = await axios.get(`https://www.themealdb.com/api/json/v1/1/search.php?f=${input}`);
      addFoods(data.meals);
      const el = document.querySelector('.error');
      el.innerHTML = '';
    } catch (err) {
      const el = document.querySelector('.error');
      el.innerHTML = 'Sorry, please try again!';
    }
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (input === '') return;
    fetchData();
    setInput('');
    e.target.firstChild.value = '';
  };

  return (
    <div>
      <form onSubmit={handleSubmit} className="form-inline filter-form">
        <input className="form-control mb-2 mr-sm-2 form-input" placeholder="Search by first letter" onChange={(e) => setInput(e.target.value)} />
        <button className="btn btn-md btn-secondary" type="submit">Search</button>
      </form>
    </div>
  );
};

FilterForm.propTypes = {
  addFoods: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  filters: state.filters,
});

const mapDispatchToProps = (dispatch) => ({
  setTitleFilter: (title) => dispatch(setTitleFilter(title)),
  addFoods: (data) => dispatch(addFoods(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FilterForm);
