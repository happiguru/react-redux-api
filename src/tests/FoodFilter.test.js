import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import FilterForm from '../containers/FilterForm';
import TestStore from './store';

describe('FoodFilterList', () => {
  test('should match with snapshot', () => {
    const jsx = (
      <Provider store={TestStore}>
        <Router>
          <FilterForm />
        </Router>
      </Provider>
    );
    const tree = (() => renderer.create(jsx).toJSON());
    expect(tree).toMatchSnapshot();
  });
});
