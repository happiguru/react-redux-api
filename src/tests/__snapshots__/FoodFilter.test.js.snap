// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodFilterList should match with snapshot 1`] = `[Function]`;
